pkg/netceptor/netceptor.go

```diff
@@ -1937,13 +1937,17 @@ func (s *Netceptor) runProtocol(ctx context.Context, sess BackendSession, bi *Ba
 						remoteCost, ok := ri.Connections[s.nodeID]
 						if !ok {
 							if remoteEstablished {
+								s.removeConnection(remoteNodeID)
+
 								return s.sendAndLogConnectionRejection(remoteNodeID, ci, "remote node no longer lists us as a connection")
```

Comment:
This function name seems like it indicates something being rejected. So maybe we don't want to delete this here?